flixkitv1_1/generator/generator.go

```diff
@@ -0,0 +1,282 @@
+
+	if len(imports) == 0 {
+		return nil
+	}
+
+	// fill in dependence information
+	g.template.Data.Dependencies = make([]v1_1.Dependency, 0)
+	for _, imp := range imports {
+		contractName, err := v1_1.ExtractContractName(imp.String())
+		if err != nil {
+			return err
+		}
+		networks, err := g.generateDependenceInfo(ctx, contractName)
+		if err != nil {
+			return err
+		}
+		c := v1_1.Contract{
+			Contract: contractName,
+			Networks: networks,
+		}
+		dep := v1_1.Dependency{
+			Contracts: []v1_1.Contract{c},
+		}
+		g.template.Data.Dependencies = append(g.template.Data.Dependencies, dep)
+	}
+
+	return nil
+}
+
+func (g *Generator) generateDependenceInfo(ctx context.Context, contractName string) ([]v1_1.Network, error) {
+	// only support string import syntax
+	contractNetworks := g.LookupImportContractInfo(contractName)
+	if len(contractNetworks) == 0 {
+		return nil, fmt.Errorf("could not find contract dependency %s", contractName)
+	}
+	var networks []v1_1.Network
+	for _, n := range contractNetworks {
+		network := v1_1.Network{
+			Network: n.Network,
+			Address: n.Address,
+		}
+		var flowkit *flowkit.Flowkit
+		if n.Network == config.MainnetNetwork.Name && g.mainnetClient != nil {
+			flowkit = g.mainnetClient
+		} else if n.Network == config.TestnetNetwork.Name && g.testnetClient != nil {
+			flowkit = g.testnetClient
+		}
+		if n.DependencyPinBlockHeight == 0 && flowkit != nil {
+			block, _ := flowkit.Gateway().GetLatestBlock()
+			height := block.Height
+
+			details, err := g.GenerateDepPinDepthFirst(ctx, flowkit, n.Address, contractName, height)
+			if err != nil {
+				return nil, err
+			}
+			network.DependencyPinBlockHeight = height
+			network.DependencyPin = details
+		}
+		networks = append(networks, network)
+	}
+
+	return networks, nil
+}
+
+func (g *Generator) LookupImportContractInfo(contractName string) []v1_1.Network {
+	for _, contract := range g.deployedContracts {
+		if contractName == contract.Contract {
+			return contract.Networks
+		}
+	}
+	return nil
+}
+
+func (g *Generator) GenerateDepPinDepthFirst(ctx context.Context, flowkit *flowkit.Flowkit, address string, name string, height uint64) (details *v1_1.PinDetail, err error) {
+	memoize := make(map[string]v1_1.PinDetail)
+	networkPinDetail, err := generateDependencyNetworks(ctx, flowkit, address, name, memoize, height)
+	if err != nil {
+		return nil, err
+	}
+
+	return networkPinDetail, nil
+}
+
+func generateDependencyNetworks(ctx context.Context, flowkit *flowkit.Flowkit, address string, name string, cache map[string]v1_1.PinDetail, height uint64) (*v1_1.PinDetail, error) {
+	identifier := fmt.Sprintf("A.%s.%s", strings.ReplaceAll(address, "0x", ""), name)
+	pinDetail, ok := cache[identifier]
+	if ok {
+		return &pinDetail, nil
+	}
+
+	account, err := flowkit.GetAccount(ctx, flow.HexToAddress(address))
+	if err != nil {
+		return nil, err
+	}
+	code := account.Contracts[name]
+	if !strings.HasPrefix(address, "0x") {
```

Comment:
i swear theres a flow library thats able to convert address to flow address which you gcan get the 0x from 🤔

Reply:
wasn't able to use Address HexWithPrefix, also noticed that flow-cli doesn't use it as well. I think it's fine